### PR TITLE
[Merged by Bors] - feat: `uIoo`

### PR DESCRIPTION
**Pull request by AlexKontorovich**
This PR is the first in a series which breaks #9598 (proving that holomorphic functions on disks have primitives) into smaller pieces. This piece introduces the notation `uIoo` (unordered Interval, open-open) which is the `oo` analog of `uIcc`. Then we prove the analogous API theorems to those available for `uIcc`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Order/Interval/Set/UnorderedInterval.lean [hunk 1]
```diff
@@ -304,6 +304,54 @@ lemma uIoc_injective_right (a : α) : Injective fun b => Ι b a := by
...
+lemma uIoo_self : uIoo a a = ∅ := by simp [uIoo]
+
+lemma Ioo_subset_uIoo : Ioo a b ⊆ uIoo a b := Ioo_subset_Ioo inf_le_left le_sup_right
+
+/-- Same as `Ioo_subset_uIoo` but with `Ioo a b` replaced by `Ioo b a`. -/
+lemma Ioo_subset_uIoo' : Ioo b a ⊆ uIoo a b := Ioo_subset_Ioo inf_le_right le_sup_left
+
+variable {x : α}
+
+lemma mem_uIoo_of_le (ha : a < x) (hb : x < b) : x ∈ uIoo a b := Ioo_subset_uIoo ⟨ha, hb⟩
```
Comment (jcommelin) [inline Mathlib/Order/Interval/Set/UnorderedInterval.lean:338]:
```suggestion
lemma mem_uIoo_of_lt (ha : a < x) (hb : x < b) : x ∈ uIoo a b := Ioo_subset_uIoo ⟨ha, hb⟩
```

File: Mathlib/Order/Interval/Set/UnorderedInterval.lean [hunk 2]
```diff
@@ -304,6 +304,54 @@ lemma uIoc_injective_right (a : α) : Injective fun b => Ι b a := by
...
+
+lemma mem_uIoo_of_ge (hb : b < x) (ha : x < a) : x ∈ uIoo a b := Ioo_subset_uIoo' ⟨hb, ha⟩
```
Comment (jcommelin) [inline Mathlib/Order/Interval/Set/UnorderedInterval.lean:340]:
```suggestion
lemma mem_uIoo_of_gt (hb : b < x) (ha : x < a) : x ∈ uIoo a b := Ioo_subset_uIoo' ⟨hb, ha⟩
```

**Review by jcommelin — COMMENTED** [review]
bors d+

**Pull request closed without merging**